corporate/lib/stripe.py

```diff
@@ -1910,21 +1924,47 @@ def process_initial_upgrade(
                 # TODO: Do a check for max licenses for fixed price plans here after we add that.
                 if (
                     stripe_invoice_paid
-                    and billed_licenses != licenses
+                    and billable_licenses != licenses
                     and not customer.exempt_from_license_number_check
                     and not fixed_price_plan_offer
                 ):
-                    # Customer paid for less licenses than they have.
-                    # We need to create a new ledger entry to track the additional licenses.
-                    LicenseLedger.objects.create(
-                        plan=plan,
-                        is_renewal=False,
-                        event_time=event_time,
-                        licenses=billed_licenses,
-                        licenses_at_next_renewal=billed_licenses,
-                    )
-                    # Creates due today invoice for additional licenses.
-                    self.invoice_plan(plan, event_time)
+                    # Billable licenses in use do not match what was paid/invoiced by customer.
+                    if billable_licenses > licenses:
+                        # Customer paid for less licenses than they have in use.
+                        # We need to create a new ledger entry to track the additional licenses.
+                        LicenseLedger.objects.create(
+                            plan=plan,
+                            is_renewal=False,
+                            event_time=event_time,
+                            licenses=billable_licenses,
+                            licenses_at_next_renewal=billable_licenses,
+                        )
+                        # Creates due today invoice for additional licenses.
+                        self.invoice_plan(plan, event_time)
+                    else:
+                        # Customer paid for more licenses than they have in use.
+                        # We need to create a new ledger entry to track the reduced renewal licenses.
+                        LicenseLedger.objects.create(
+                            plan=plan,
+                            is_renewal=False,
+                            event_time=event_time,
+                            licenses=licenses,
+                            licenses_at_next_renewal=billable_licenses,
+                        )
+                        # Send internal billing notice about license discrepancy.
```

Comment:
We've already done an initial Stripe payment/charge in this case, so we'll want to do some manual follow-up and investigation here. But the check in `do_upgrade` makes this much less likely to happen.